tests/comestible_test.cpp

```diff
@@ -178,3 +179,83 @@ TEST_CASE( "cooked_veggies_get_correct_calorie_prediction", "[recipe]" )
     CHECK( default_nutrition.kcal == predicted_nutrition.first.kcal );
     CHECK( default_nutrition.kcal == predicted_nutrition.second.kcal );
 }
+
+// The Character::compute_effective_food_volume_ratio function returns a floating-point ratio
+// used as a multiplier for the food volume when it is eaten, based on the energy density
+// (kcal/gram) of the food, as follows:
+//
+// - low-energy food     (0.0 < kcal/gram < 1.0)  returns 1.0
+// - medium-energy food  (1.0 < kcal/gram < 3.0)  returns (kcal/gram)
+// - high-energy food    (3.0 < kcal/gram)        returns sqrt( 3 * kcal/gram )
+//
+// The Character::compute_calories_per_effective_volume function returns a dimensionless integer
+// representing the "satiety" of the food, with higher numbers being more calorie-dense, and lower
+// numbers being less so.
+//
+TEST_CASE( "effective food volume and satiety", "[character][food][satiety]" )
+{
+    const Character &u = get_player_character();
+    double expect_ratio;
+
+    // Apple: 95 kcal / 200 g (1 serving)
+    const item apple( "test_apple" );
+    const nutrients apple_nutr = u.compute_effective_nutrients( apple );
+    REQUIRE( apple.count() == 1 );
+    REQUIRE( apple.weight() == 200_gram );
+    REQUIRE( apple.volume() == 250_ml );
+    REQUIRE( apple_nutr.kcal == 95 );
+    // If kcal per gram < 1.0, return 1.0
+    CHECK( u.compute_effective_food_volume_ratio( apple ) == Approx( 1.0f ).margin( 0.01f ) );
+    CHECK( u.compute_calories_per_effective_volume( apple ) == 396 );
+    CHECK( satiety_bar( 396 ) == "<color_c_yellow>||</color>..." );
```

Comment:
```suggestion
    // NOLINTNEXTLINE(cata-text-style): verbatim ellipses necessary for validation
    CHECK( satiety_bar( 396 ) == "<color_c_yellow>||</color>..." );
```

this should suppress the clang-tidy ellipses warning.